ansible/roles/lib_openshift_3.2/build/src/oc_image.py

```diff
@@ -0,0 +1,41 @@
+# pylint: skip-file
+
+# pylint: disable=too-many-arguments
+class OCImage(OpenShiftCLI):
+    ''' Class to wrap the oc command line tools
+    '''
+    def __init__(self,
+                 namespace,
+                 registry_url,
+                 image_name,
+                 image_tag,
+                 kubeconfig='/etc/origin/master/admin.kubeconfig',
+                 verbose=False):
+        ''' Constructor for OpenshiftOC '''
+        super(OCImage, self).__init__(namespace, kubeconfig)
+        self.namespace = namespace
+        self.registry_url = registry_url
+        self.image_name = image_name
+        self.image_tag = image_tag
+        self.kubeconfig = kubeconfig
+        self.verbose = verbose
+
+    def get(self):
+        '''return a image by name '''
+        results = self._get('imagestream', self.image_name)
+        results['exists'] = False
+        if results['returncode'] == 0 and results['results'][0]:
+            results['exists'] = True
+
+        if results['returncode'] != 0 and '"%s" not found' % self.image_name in results['stderr']:
+            results['returncode'] = 0
+
+        return results
+
+    def create(self, url=None, name=None, tag=None):
+        '''Create an image '''
+
+        image = '{0}:{1}'.format(name, tag)
+        cmd = [image, '--from={0}/{1}'.format(url, image), '-n{0}'.format(self.namespace), '--confirm']
```

Comment:
I'd prefer it if this logic was moved down inside of the _import_image call.  That way I can call import_image from another module if needed.

_import_image would take ` url=None, name=None, tag=None`